core/domain/state_domain.py

```diff
@@ -1925,6 +1928,306 @@ def validate(self) -> None:
                 self.duration_secs)
 
 
+def validate_rte_tags(
+    html: str, is_tags_nested_inside_tabs_or_collapsible: bool = False
+) -> None:
+    """Validate all the RTE tags.
+
+    Args:
+        html: str. The RTE content of the state.
+        is_tags_nested_inside_tabs_or_collapsible: bool. True when we
+            validate tags inside `Tabs` or `Collapsible` tag.
+
+    Raises:
+        utils.ValidationError. Raises the validation error in case the
+            content is not valid.
+    """
+    soup = bs4.BeautifulSoup(html, 'html.parser')
+    empty_values = ['&quot;&quot;', '', '\'\'', '\"\"']
+
+    for tag in soup.find_all('oppia-noninteractive-image'):
+        if not tag.has_attr('alt-with-value'):
+            raise utils.ValidationError(
+                'Image tag does not have \'alt-with-value\' attribute.'
+            )
+        if '&quot;' in tag['alt-with-value']:
+            if len(tag['alt-with-value'].strip().split('&quot;')[1]) < 5:
+                raise utils.ValidationError(
+                    'The length of the image tag \'alt-with-value\' '
+                    'attribute value should be at least 5 characters.'
+                )
+        else:
+            if len(tag['alt-with-value'].strip()) < 5:
+                raise utils.ValidationError(
+                    'The length of the image tag \'alt-with-value\' '
+                    'attribute value should be at least 5 characters.'
+                )
```

Comment:
Why are there 2 possibilities for this validation? When does it have `'&quot;'` and when does it not have it?

Reply:
To pass the tests present inside our codebase. In some places we use something like - `alt-with-value="image"` and at some places we use `alt-with-value="&ampquot;image&ampquot;"`(this is the correct one). So just to avoid them I have added this check too so that I don't mess up with the original test.
Thanks

Reply:
Also, in case you missed my comment [here](https://github.com/oppia/oppia/pull/16065#issuecomment-1257090080). Thanks

Reply:
Hm I feel like this is not correct. Can you create an image and add an `alt-with-value` of `hello` versus an `alt-with-value` of `"hello"` and see what the difference is? We need to make sure that when the user enters a quotation mark as part of the alt caption, we aren't treating it as special (it should still just be a single character). And when you get the result of that, feel free to fix this if needed and update me!

Reply:
Yep! You were right, that code wouldn't work properly when we have value with `""`. I have updated the code now and have tested it with various values. Please let me know if any further changes are required. Just to give some context, when we have value like `"image"` then we will receive `\"&amp;quot;\\&amp;quot;image\\&amp;quot;&amp;quot;\` in backend and if we have a simple value like `image` then we will get `\"&amp;quot;image&amp;quot;\`. According to my  new code we will be able to get the exact value and test it.
Thanks

Reply:
This is your new code:
```
alt_value = utils.unescape_html(
  tag['alt-with-value']
)[1:-1].replace('\\"', '')
```

Let's say you have `image` as the alt text. Here, the `tag` variable then has value `\"&amp;quot;image&amp;quot;\`. After calling `unescape_html` on it, you get `\"&quot;image&quot;\`. Then you do `[1:-1]` to get `"&quot;image&quot;` and `replace` does nothing. But that doesn't end up at `image` like it should.

Look at how the RTE tags are converted from the backend storage `\"&amp;quot;image&amp;quot;\` all the way to the frontend UI `image`. You just need to duplicate that code to get the right string.

Reply:
Oh, that is not the case here. When we call `unescape_html` we will get `"image"` instead of `\"&quot;image&quot;\` because firstly `&amp` will be converted to `&` and then `&quot;` will be converted to `"`

Reply:
Ah I see -- thanks for the clarification!